test/make_test_files.py

```diff
@@ -12,11 +12,13 @@
 f.create_dataset("string_arr", (2,), dtype=sarray_type)
 f["string_arr"][:] = np.array([b"hello", b"there"] * 2 * 2).reshape((2,2,2))
 
-f.create_dataset("compound", (2,), dtype=[('floaty', array_type), ('stringy',sarray_type)])
-f['compound'][:] = np.array([(f['float_arr'][0], f['string_arr'][0]), (f['float_arr'][1], f['string_arr'][1])], dtype=f['compound'].dtype)
+f.create_dataset("compound", (2,), dtype=[("floaty", array_type), ("stringy",sarray_type)])
+f["compound"][:] = np.array([(f["float_arr"][0], f["string_arr"][0]), (f["float_arr"][1], f["string_arr"][1])], dtype=f["compound"].dtype)
 
-f.create_dataset("bool", data=[False, True, True, False], shape=(2,2))
+f.create_dataset("bool", data=[[False, True], [True, False]], shape=(2,2))
```

Comment:
Made the `data` match the `shape`

Reply:
If it passes the tests still, sounds good!

Reply:
Yeah, in the end, I don't think it was necessary :sweat_smile: 

Well, at least, the passing tests show that `.value` and `.to_array` are robust enough